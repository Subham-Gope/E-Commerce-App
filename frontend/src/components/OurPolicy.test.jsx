// import { describe, test, expect, vi } from "vitest";
// import { render, screen } from "@testing-library/react";
// import OurPolicy from "./OurPolicy";

// // Mocking assets import
// vi.mock("../assets/assets", () => ({
//   assets: {
//     exchange_icon: "exchange_icon.png",
//     quality_icon: "quality_icon.png",
//     support_img: "support_img.png",
//   },
// }));

// describe("Testing the OurPolicy.jsx file", () => {
//   test("rendering the texts", () => {
//     // rendering and setting up the snapshot matching
//     const { container } = render(<OurPolicy />);
//     expect(container).toMatchSnapshot();

//     //checking the headings
//     expect(screen.getByText(/easy exchange policy/i)).toBeInTheDocument();
//     expect(screen.getByText(/7 days return policy/i)).toBeInTheDocument();
//     expect(screen.getByText(/best customer support/i)).toBeInTheDocument();

//     // checking descriptions
//     expect(
//       screen.getByText(/we offer hassle free exchange policy/i)
//     ).toBeInTheDocument();
//     expect(
//       screen.getByText(/We provide 7 days of free return policy/i)
//     ).toBeInTheDocument();

//     expect(
//       screen.getByText(/we provide 24\/7 customer support/i)
//     ).toBeInTheDocument();

//     // checking images
//     const images = screen.getAllByAltText(/icon/i);
//     expect(images).toHaveLength(3);

//     expect(images[0]).toHaveAttribute("src", "exchange_icon.png");
//     expect(images[1]).toHaveAttribute("src", "quality_icon.png");
//     expect(images[2]).toHaveAttribute("src", "support_img.png");
//   });
// });
