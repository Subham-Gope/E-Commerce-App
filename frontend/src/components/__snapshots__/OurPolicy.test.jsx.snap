// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the OurPolicy.jsx file > rendering the texts 1`] = `
<div>
  <div
    class="flex flex-col sm:flex-row justify-around gap-12 sm:gap-2 text-center py-20 text-xs sm:text-sm md:text-base text-gray-700"
  >
    <div>
      <img
        alt="exchange icon"
        class="w-12 m-auto mb-5"
        src="exchange_icon.png"
      />
      <p
        class="font-semibold"
      >
        Easy Exchange Policy
      </p>
      <p>
        We Offer hassle free exchange policy
      </p>
    </div>
    <div>
      <img
        alt="quality icon"
        class="w-12 m-auto mb-5"
        src="quality_icon.png"
      />
      <p
        class="font-semibold"
      >
        7 Days Return Policy
      </p>
      <p>
        We provide 7 days of free return policy
      </p>
    </div>
    <div>
      <img
        alt="support icon"
        class="w-12 m-auto mb-5"
        src="support_img.png"
      />
      <p
        class="font-semibold"
      >
        Best customer support
      </p>
      <p>
        we provide 24/7 customer support
      </p>
    </div>
  </div>
</div>
`;
